Libraries/LibWeb/XPath/XPathResult.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2025, Johannes Gustafsson <johannesgu@outlook.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/String.h>
+#include <AK/Vector.h>
+#include <LibWeb/Bindings/PlatformObject.h>
+#include <LibWeb/Forward.h>
+#include <LibWeb/WebIDL/Types.h>
+
+namespace Web::XPath {
+
+class XPathResult : public Bindings::PlatformObject {
+    WEB_PLATFORM_OBJECT(XPathResult, Bindings::PlatformObject);
+    GC_DECLARE_ALLOCATOR(XPathResult);
+
+public:
+    unsigned short const ANY_TYPE = 0;
+    unsigned short const NUMBER_TYPE = 1;
+    unsigned short const STRING_TYPE = 2;
+    unsigned short const BOOLEAN_TYPE = 3;
+    unsigned short const UNORDERED_NODE_ITERATOR_TYPE = 4;
+    unsigned short const ORDERED_NODE_ITERATOR_TYPE = 5;
+    unsigned short const UNORDERED_NODE_SNAPSHOT_TYPE = 6;
+    unsigned short const ORDERED_NODE_SNAPSHOT_TYPE = 7;
+    unsigned short const ANY_UNORDERED_NODE_TYPE = 8;
+    unsigned short const FIRST_ORDERED_NODE_TYPE = 9;
```

Comment:
Types related to web specs often use the WebIDL types, i.e. `WebIDL::UnsignedShort` instead of `unsigned short`. For all other code, we usually use the types as defined in `AK/Types.h`, i.e. `u16`.

Same for the fields `m_result_type` and `m_snapshot_length`.